app/src/lib/token-utils.js

```diff
@@ -33,7 +33,7 @@ export async function createTokenEVMScript(accounts, tokenManagerAddress) {
  * @returns {Array<string>} array with the [<whole>, <decimal>] parts of the number
  */
 function splitDecimalNumber(num) {
-  const [whole = '', dec = ''] = num.split('.')
+  const [whole = '', dec = ''] = String(num).split('.')
```

Comment:
We should receive a string. We should trace why we receive a number here.

Reply:
values comes directly from input[number] here, maybe we can cast earlier ? or use decimals.js 😇

Reply:
Ok, if there is no conversion to number in the meanwhile we are probably fine. I would cast the number to string just as you are doing it.